### DISCUSSION TITLE
Counter tags not attached

### DISCUSSION THREAD

**Discussion by YarinOmesi** [open-telemetry/opentelemetry-dotnet#5313] [Q&A] [answered]
System.Diagnostics.DiagnosticSource v8.0 introduced the ability to instantiate a [Counter with tags](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.metrics.meter.createcounter?view=net-8.0#system-diagnostics-metrics-meter-createcounter-1(system-string-system-string-system-string-system-collections-generic-ienumerable((system-collections-generic-keyvaluepair((system-string-system-object)))))). 

But when tried to instantiate a Counter with tags, it's tags were not attached to the measurement, both in console and prometheus exporters.

I have seen a similar issue with meter tags, but none with counter tags.

This is how it shoud be ? or is it a bug ?

The following exampele use version `OpenTelemetry 1.7.0` and `.net 8`

### Example Code
```csharp 
Meter meter = new Meter("MyMeter", "1.0");

TagList counterTags = new TagList() {
    {"counter_tag", true}
};

Counter<long> myCounter = meter.CreateCounter<long>("myCounter", null, null, counterTags);

MeterProvider meterProvider = Sdk.CreateMeterProviderBuilder()
    .AddMeter("MyMeter")
    .AddConsoleExporter()
    .AddPrometheusHttpListener(options => options.UriPrefixes = new []{"http://localhost:9000"})
    .Build();

myCounter.Add(1);

meterProvider.Dispose();
```

### Console Output
```
Resource associated with Metric:
    telemetry.sdk.name: opentelemetry                               
    telemetry.sdk.language: dotnet                                  
    telemetry.sdk.version: 1.7.0                                    
    service.name: unknown_service:MinimalDemo                       
                                                                    
Metric Name: myCounter, Meter: MyMeter/1.0                          
(2024-02-03T16:35:29.2264735Z, 2024-02-03T16:35:29.2324484Z] LongSum
Value: 1 
```

### Prometheus Output
```
# TYPE myCounter_total counter
myCounter_total 1 1706978095318

# EOF
```

**Comment by cijothomas** [comment, marked as answer]
Counter tags (and other instrument tag), is not implemented in OTel yet. It is not something strictly from the spec, so we are working to find out how best to handle. There is a good chance it'll used to support the notion of "bound instruments", but nothing solid yet.

Meter tags are now supported on Console/OTLPExporters (its called InstrumentationScope in OTLP)
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Console/CHANGELOG.md#170-rc1

**Comment by YarinOmesi** [comment]
Any new on this ? 

will it be better if I open an issue on this ?

**Reply by cijothomas** [reply]
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/metrics#metrics-enrichment
yes. you can open a new issue, for better tracking.
